neqo-http3/src/connection_client.rs

```diff
@@ -854,13 +854,38 @@ impl Http3Client {
             .stats(&mut self.conn)
     }
 
-    /// This function combines  `process_input` and `process_output` function.
-    pub fn process(&mut self, dgram: Option<&Datagram>, now: Instant) -> Output {
+    pub fn process_into_buffer<'a>(
+        &mut self,
+        input: Option<Datagram<&[u8]>>,
+        now: Instant,
+        out: &'a mut Vec<u8>,
+    ) -> Output<&'a [u8]> {
         qtrace!([self], "Process.");
-        if let Some(d) = dgram {
+        if let Some(d) = input {
             self.process_input(d, now);
+        } else {
+            // TODO: Optimization worth it? Given that process_input calls
+            // process_http3, moving the line below outside of conditional makes
+            // it run twice when input is some.
+            self.process_http3(now);
```

Comment:
Is there still anything left to be done about this `TODO`? Because if it works, why not just do it?

Reply:
Addressed in https://github.com/mozilla/neqo/pull/2093/commits/15eb2f8abbab2e4d6536973c64ffedc1f8c1b612.